lib/handlers/compile.js

```diff
@@ -131,7 +131,15 @@ export class CompileHandler {
         const isPrediscovered = !!compiler.version;
 
         const type = compiler.compilerType || 'default';
-        const compilerClass = getCompilerTypeByKey(type);
+        let compilerClass;
+        try {
+            compilerClass = getCompilerTypeByKey(type);
+        } catch (e) {
+            logger.error(`Compiler ID: ${compiler.id}`);
+            logger.error(e);
+            process.exit(1);
```

Comment:
I think 1 is already in use, maybe it should be something else

Reply:
I don't think we do, I can't find any